modules/javafx.graphics/src/main/native-prism-mtl/MetalPipelineManager.m

```diff
@@ -0,0 +1,366 @@
+/*
+ * Copyright (c) 2021, 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+#import "MetalPipelineManager.h"
+#import "MetalRTTexture.h"
+#include "com_sun_prism_mtl_MTLContext.h"
+
+// ---------------------------- Debug helper for Developers -------------------------
+//
+// This implementation is to utilize "Metal Debuger" present in Xcode.
```

Comment:
is the debugger available in production code?

Reply:
It is available to developers only. It requires two things
Un-comment line 51: //#define JFX_MTL_DEBUG_CAPTURE
and enable metal debug trace capturing, on terminal, export MTL_CAPTURE_ENABLED=1